### PR TITLE
Platform designer update 2023.2

### PR DESCRIPTION
**Pull request by whitepau**
# Existing Sample Changes
## Description

Update code sample to work with 2023.2:

1. No longer need to set `reset_n` in Platform Designer
2. update TCL test to be more modular
3. No support for simulating the `avalon_mm` protocol, so use `avalon_mm_uses_ready` for simulation only.

Fixes Issue# 

## External Dependencies

none

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Implement fixes for ONSAM Jiras

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [x] VSCode
- [x] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

Regression tests:

Linux: https://spetc.intel.com/testanalysis?trview=0&testRunIds=8136788&tapgsize=20&tasort=testCasePath&tasortdir=asc
Windows: https://spetc.intel.com/testanalysis?trview=0&testRunIds=8136792&tapgsize=20&tasort=testCasePath&tasortdir=asc

### CODE REVIEW

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-quartus-sln/.gitignore [hunk 1]
```diff
@@ -10,7 +10,7 @@ add.qws
 add_kernel_wrapper
 
 # ignore oneAPI generated IP
-add.fpga_ip_export.prj_1
+*.prj
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-quartus-sln/.gitignore:13]: Please remove this .gitignore file from the PR

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-quartus-sln/add_kernel_wrapper.qsys [hunk 2]
```diff
@@ -33,14 +33,14 @@ https://fpgasoftware.intel.com/eula.-->
         <ipxact:parameter parameterId="board" type="string">
           <ipxact:name>board</ipxact:name>
           <ipxact:displayName>Board</ipxact:displayName>
-          <ipxact:value>default</ipxact:value>
+          <ipxact:value>Unknown</ipxact:value>
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-quartus-sln/add_kernel_wrapper.qsys:36]: Is this expected?

Comment (whitepau) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-quartus-sln/add_kernel_wrapper.qsys:36]: Yes. don't ask me why it's expected  though; i don't know why it works, just that it compiles with Intel® Quartus® Prime 22.2 and 23.1 :)

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+add-quartus/
```
[the remark this answers is not in the document]
Comment (whitepau) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/.gitignore]: done

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-quartus-sln/.gitignore (hunk 1 above)
Comment (whitepau) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-quartus-sln/.gitignore:13]: done

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**